core/server/worker/src/main/java/alluxio/worker/dora/RocksDBDoraMetaStore.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.dora;
+
+import alluxio.proto.meta.DoraMeta;
+import alluxio.rocks.RocksStore;
+import alluxio.util.io.PathUtils;
+
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.ColumnFamilyOptions;
+import org.rocksdb.CompressionType;
+import org.rocksdb.DBOptions;
+import org.rocksdb.HashLinkedListMemTableConfig;
+import org.rocksdb.ReadOptions;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.RocksObject;
+import org.rocksdb.WriteOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+/**
+ * Dora Meta Store in RocksDB.
+ */
+public class RocksDBDoraMetaStore implements DoraMetaStore {
+  private static final Logger LOG = LoggerFactory.getLogger(RocksDBDoraMetaStore.class);
+  private static final String DORA_META_DB_NAME = "DoraMeta";
+  private static final String DORA_META_FILE_STATUS_COLUMN = "FileStatusCF";
+  private static final String DORA_META_STORE_NAME = "DoraMetaStore";
+
+  // These are fields instead of constants because they depend on the call to RocksDB.loadLibrary().
+  private final WriteOptions mWriteOption;
+  private final ReadOptions mReadOption;
+  private final RocksStore mRocksStore;
+  private final List<RocksObject> mToClose = new ArrayList<>();
+
+  private final AtomicReference<ColumnFamilyHandle> mFileStatusColumn = new AtomicReference<>();
+
+  /**
+   * Queries the metadata for a file.
+   *
+   * @param path the full path of this file
+   * @return the metadata if found, Optional.empty if not found
+   */
+  @Override
+  public Optional<DoraMeta.FileStatus> getDoraMeta(String path) {
+    byte[] status;
+    try {
+      status = db().get(mFileStatusColumn.get(), path.getBytes());
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+    if (status == null) {
+      return Optional.empty();
+    }
+    try {
+      return Optional.of(DoraMeta.FileStatus.parseFrom(status));
```

Comment:
This is kind of disk/memory data structure incompatibility.
When we doing upgrade, we need to make sure the upgrade can recognize the old data format.
If not, this is a disruptive upgrade.

I am not sure but it seem the grpc.proto can handle some level of data structure change, because we define all the members as optional.